### PR TITLE
[Cling] Update cpt to build Cling using LLVM 9

### PR DESCRIPTION
**Pull request by reikdas**
Only builds using LLVM 9 source works right now (using `--with-binary-llvm` and `--with-llvm-tar` flags will cause a build error).

A single test fails. Reproducer - `cat <path to cling source dir>/test/Lookup/template.C | cling --nologo -I<path to build dir>/tools/clang/include -Xclang -verify -fno-rtti 2>&1`

### CODE REVIEW

File: interpreter/cling/lib/Interpreter/ValuePrinter.cpp [hunk 1]
```diff
@@ -549,7 +549,7 @@ static const char* BuildAndEmitVPWrapperBody(cling::Interpreter &Interp,
                                              clang::QualType QT,
                                              const void* ValPtr)
 {
-  const clang::SourceLocation noSrcLoc;
+  const clang::SourceLocation noSrcLoc = {};
```
Comment (Axel-Naumann) [inline interpreter/cling/lib/Interpreter/ValuePrinter.cpp:552]: Why is this needed?

Comment (reikdas) [reply interpreter/cling/lib/Interpreter/ValuePrinter.cpp:552]: It looks like very old compilers like Clang 3.8 do not support default initialization of a `const` object, without an explicitly specified default constructor - https://github.com/reikdas/cling/runs/2227072695?check_suite_focus=true#step:6:2897

Comment (Axel-Naumann) [reply interpreter/cling/lib/Interpreter/ValuePrinter.cpp:552]: Clang 3.8 is historical. Can we just not support that one, please?

Comment (vgvassilev) [reply interpreter/cling/lib/Interpreter/ValuePrinter.cpp:552]: That'd mean dropping a build that we supported. If that's the only change why not keeping it and the Ubu16 platform?

Comment (Axel-Naumann) [reply interpreter/cling/lib/Interpreter/ValuePrinter.cpp:552]: Because any single change is "the only change" and we will accumulate "single changes" at the cost of baroque code. clang 3.8 is dead old and this is a compiler bug; let's drop support. Coverage matters a ton when it comes to GCC 10, llvm 12 etc.

Comment (reikdas) [reply interpreter/cling/lib/Interpreter/ValuePrinter.cpp]: Updated the PR to drop Ubuntu 16.04 Clang build, and revert this change.

**Review by Axel-Naumann — APPROVED** [review]
